### PR TITLE
BC-11668 - Check and fix CVEs and update libs

### PR DESCRIPTION
**Pull request by uidp**
# Description

Highlights:
- **CKEditor5** v36 → v41.4.2 (switched to individual `@ckeditor/ckeditor5-*` packages; math plugin replaced with `@isaul32/ckeditor5-math@41.4.2`)
- **webpack** 4 → 5 (filesystem cache in dev, production mode via `WEBPACK_PRODUCTION` env var)
- **MathJax** replaced by KaTeX
- **gulp-cli** upgraded to `^3.1.0`
- **`NODE_OPTIONS=--openssl-legacy-provider`** removed (no longer needed with webpack 5 + Node 24)
- **7 high-severity CVEs fixed** via npm `overrides` (`tar`, `serialize-javascript`, `braces`, `micromatch`, `lodash-es`, `chokidar`, `cacache`)

## Links to Tickets or other pull requests
BC-11668

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,33 +59,33 @@
...
 		"i18next-localstorage-backend": "^3.1.3",
-		"i18next-sync-fs-backend": "^1.1.1",
 		"iovalkey": "^0.3.1",
 		"jquery": "^3.6.0",
 		"jquery-datetimepicker": "^2.5.21",
 		"jsonwebtoken": "^9.0.3",
-		"lodash": "^4.17.23",
-		"mathjax": "^2.7.9",
+		"lodash": "^4.18.1",
+		"mathjax": "^4.1.2",
```
Comment (virgilchiriac) [inline package.json]: maybe easier to have simply upgraded to v3, and not having the need to add the patch

It sais that v4 contains more additions and not breaking changes, but "There are a number of changes to the TeX input jax and its extensions that may affect existing content" which might be more difficult to test.
https://docs.mathjax.org/en/latest/upgrading/v3.html

**Review by virgilchiriac — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -2,52 +2,53 @@
...
-		"watch": "node --watch ./bin/www",
+		"watch": "concurrently --names \"server,assets\" --prefix-colors \"cyan,yellow\" \"node --unhandled-rejections=warn --watch ./bin/www\" \"gulp watch\"",
 		"debug": "node --inspect=9310 --watch ./bin/www",
 		"lint": "eslint --ext .json --fix . && eslint ./controllers ./helpers ./test/mocha ./api.js ./app.js ./locales --ext .js --ext .json --fix",
 		"mocha": "cross-env NODE_ENV=test mocha test/mocha/ --exit --recursive",
-		"build": "gulp clear && cross-env NODE_OPTIONS=--openssl-legacy-provider gulp",
-		"coverage": "nyc npm run mocha"
+		"build": "gulp clear && gulp",
+		"coverage": "nyc npm run mocha",
+		"postinstall": "patch-package"
```
Comment (virgilchiriac) [inline package.json]: is this still needed?

File: patches/@isaul32+ckeditor5-math+41.4.2.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
++++ b/node_modules/@isaul32/ckeditor5-math/src/utils.js
+@@ -13,7 +13,7 @@ export function getSelectedMathModelWidget(selection) {
+ export function isMathJaxVersion3(MathJax) {
+     return (MathJax != null && typeof MathJax == 'object' && 'version' in MathJax && typeof MathJax.version == 'string' &&
+         MathJax.version.split('.').length === 3 &&
+-        MathJax.version.split('.')[0] === '3');
++        ['3', '4'].includes(MathJax.version.split('.')[0]));
+ }
+ // Simple MathJax 2 version check
+ export function isMathJaxVersion2(MathJax) {
```
Comment (virgilchiriac) [inline patches/@isaul32+ckeditor5-math+41.4.2.patch]: is this still needed?

**Review by virgilchiriac — CHANGES_REQUESTED** [review]

**Comment by virgilchiriac** [comment]
PR description needs to be updated to mention Katex replacment for Mathjax

File: package.json (hunk 1 above)
Comment (uidp) [reply package.json]: I switched to KaTex rendering like we have it in Vue Client. Removed the Mathjax dependency.

File: package.json (hunk 2 above)
Comment (uidp) [reply package.json]: Removed

File: patches/@isaul32+ckeditor5-math+41.4.2.patch (hunk 3 above)
Comment (uidp) [reply patches/@isaul32+ckeditor5-math+41.4.2.patch]: Removed

File: ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2 [hunk 4]
```diff
@@ -35,10 +35,12 @@ data:
...
-            fontSrc: `'self' data: `,
+            fontSrc: `'self' data:`,
             styleSrc: `'self' 'unsafe-inline'`,
             // scriptSrc: "'strict-dynamic' 'unsafe-eval' 'nonce-<nonceValue>'",
             scriptSrc: `'self' 'unsafe-eval'`,
+            workerSrc: `'self' blob:`,
+            connectSrc: `'self' data: blob: {{TSP_API_BASE_NO_WWW_URL | default("") }}  {{NOT_AUTHENTICATED_REDIRECT_URL | default("") }} https://{{ DOMAIN }} wss://{{ DOMAIN }} https://api.{{ DOMAIN }} https://embed.{{ DOMAIN }} https://oauth.{{ DOMAIN }} https://{{STORAGE_PREFIX}}{{ DOMAIN }} https://etherpad.{{ DOMAIN }} https://blog.niedersachsen.cloud https://blog.dbildungscloud.de https://docs.dbildungscloud.de https://sc-content-resources.schul-cloud.org https://sc-content-resources.hpi-schul-cloud.de https://open.hpi.de https://s3.hidrive.strato.com https://scalelite.bbb.messenger.schule {% if NAMESPACE == "niedersachsen" %}https://portfolio.niedersachsen.cloud{% endif %}`,
```
Comment (virgilchiriac) [inline ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2]: looks identical with defaultSrc, in which case is not needed as defaultSrc will serve as fallback

File: config/http-headers.js
Comment (virgilchiriac) [inline config/http-headers.js:50]: just noticed that dashboard is now migrated, I think this can be safe to remove. In any case, the urls look like not valid anymore. But maybe out of scope

File: config/http-headers.js [hunk 5]
```diff
@@ -25,6 +25,8 @@ const config = {
 			fontSrc: '\'self\' data:',
 			styleSrc: '\'self\' \'unsafe-inline\'',
 			scriptSrc: '\'self\' \'unsafe-eval\'',
+			workerSrc: '\'self\' blob:',
+			connectSrc: '\'self\' data: blob: wss://dbildungscloud.de wss://scchat.dbildungscloud.de https://api.dbildungscloud.de https://scchat.dbildungscloud.de https://s3.hidrive.strato.com https://docs.dbildungscloud.de https://etherpad.dbildungscloud.de https://blog.niedersachsen.cloud https://blog.dbildungscloud.de https://sc-content-resources.hpi-schul-cloud.de https://open.hpi.de https://upload.wikimedia.org webpack:',
```
Comment (virgilchiriac) [inline config/http-headers.js]:
```
const sharedSources = '\'self\' data: blob: wss://dbildungscloud.de wss://scchat.dbildungscloud.de https://api.dbildungscloud.de https://scchat.dbildungscloud.de https://s3.hidrive.strato.com https://docs.dbildungscloud.de https://etherpad.dbildungscloud.de https://blog.niedersachsen.cloud https://blog.dbildungscloud.de https://sc-content-resources.hpi-schul-cloud.de https://open.hpi.de https://upload.wikimedia.org';

const corsDefault = {
	defaultSrc: sharedSources,
	fontSrc: '\'self\' data:',
	styleSrc: '\'self\' \'unsafe-inline\'',
	scriptSrc: '\'self\' \'unsafe-eval\'',
	workerSrc: '\'self\' blob:',
	connectSrc: `${sharedSources} webpack:`,
};
```

File: Dockerfile [hunk 6]
```diff
@@ -20,7 +20,7 @@ RUN git config --global url."https://github.com/".insteadOf git@github.com: \
 WORKDIR /app
 
 COPY package.json package-lock.json ./
-RUN npm ci --ignore-scripts
+RUN npm ci
```
Comment (virgilchiriac) [inline Dockerfile]: as a rule we keep --ignore-scripts in Docker npm ci

**Review by virgilchiriac — CHANGES_REQUESTED** [review]

File: webpack.config.js [hunk 7]
```diff
@@ -16,46 +32,30 @@ const plugins = [
...
-			// In addition the @babel/plugin-proposal-export-namespace-from
-			// was also added for that reason as well and can be removed
-			// once the update to webpack 5.x is done.
+			// All files that end on .js or .jsx are transpiled by babel.
+			// htmlparser2 and @isaul32 are excluded from the node_modules
+			// exclusion because they use syntax that requires transpilation.
 			{
 				test: /\.(?:js|jsx|cjs)$/,
-				exclude: /(node_modules)[/\\](?!(htmlparser2)[/\\])/,
+				exclude: /(node_modules)[/\\](?!(htmlparser2|@isaul32)[/\\])/,
```
Comment (virgilchiriac) [inline webpack.config.js:45]: why isaul32 here? was this not related to the patch?

File: ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2 (hunk 4 above)
Comment (uidp) [reply ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2]: Yes not needed for production. Fixed.

File: config/http-headers.js (hunk 5 above)
Comment (uidp) [reply config/http-headers.js]: Fixed.

File: Dockerfile (hunk 6 above)
Comment (uidp) [reply Dockerfile]: Fixed.

File: webpack.config.js (hunk 7 above)
Comment (uidp) [reply webpack.config.js:45]: The package `@isaul32/ckeditor5-math` uses ES import syntax. The exclusion override is still needed so webpack/babel processes it. This is independent of the patch.

**Review by virgilchiriac — APPROVED** [review]
